forum/users/tests/test_token_serializer.py

```diff
@@ -0,0 +1,123 @@
+from unittest.mock import Mock
+from django.contrib.auth import get_user_model
+from rest_framework.test import APITestCase
+from rest_framework.exceptions import ValidationError
+from rest_framework_simplejwt.tokens import RefreshToken
+from users.models import Role
+from users.serializers import CustomTokenObtainPairSerializer
+
+User = get_user_model()
+
+
+class TestCustomTokenObtainPairSerializer(APITestCase):
+    def setUp(self):
+        # Create a test user
+        self.user = User.objects.create_user(
+            email="testuser@example.com",
+            password="securepassword",
+            role=Role.INVESTOR.value  # Assign a default role
+        )
```

Comment:
Consider using `setUpTestData` instead of `setUp` if the test data does not change between tests. This can improve performance by creating the test data once for the entire TestCase.
 ```
@classmethod
def setUpTestData(cls):
    cls.user = User.objects.create_user(
        email=TEST_EMAIL,
        password=TEST_PASSWORD,
        role=Role.INVESTOR.value
    )
```